torch/_functorch/aot_autograd.py

```diff
@@ -2255,6 +2288,77 @@ def runtime_wrapper(*args):
             return fw_outs
     return runtime_wrapper
 
+@contextmanager
+def patch(module, attr, patch_fn):
+    """
+    Context manager that temporarily patches attribute of a module to a given patch_fn
+    """
+    old_fn = getattr(module, attr)
+    setattr(module, attr, patch_fn)
+    try:
+        yield
+    finally:
+        setattr(module, attr, old_fn)
+
+def create_functionalized_rng_ops_wrapper(func, args, trace_joint=True):
+    # Functionalization of rng ops changes the calling convention of the joint graph.
+    # It goes from (primals, tangents) to (seed, offset, primals, tangents)
+    # At runtime, we pass on the current seed and offset. This is hidden from
+    # the user.
+    fake_mode_init = False
+    for x in args:
+        if isinstance(x, FakeTensor):
+            fake_mode = x.fake_mode
+            fake_mode_init = True
+            break
+
+    if not fake_mode_init:
+        fake_mode = (
+            FakeTensorMode(shape_env=None)
+            if config.use_fake_tensor
+            else nullcontext()
+        )
+
+    def fakify(x):
+        return fake_mode.from_tensor(x, static_shapes=True)
+
+
+    def override_get_rng_state(device: Union[int, str, torch.device] = 'cuda'):
+        out = PhiloxStateTracker.get_state_as_tensor()
+        return out
+
+    def override_set_rng_state(x, device: Union[int, str, torch.device] = 'cuda'):
+        PhiloxStateTracker.set_state_from_tensor(x)
+
+
+    def traced_joint(fwd_seed, fwd_base_offset, bwd_seed, bwd_base_offset, primals, tangents):
+        with patch(torch.cuda, "get_rng_state", override_get_rng_state):
+            with patch(torch.cuda, "set_rng_state", override_set_rng_state):
+                return func(primals, tangents)
+
+    def traced_forward(fwd_seed, fwd_base_offset, *primals):
+        with patch(torch.cuda, "get_rng_state", override_get_rng_state):
+            with patch(torch.cuda, "set_rng_state", override_set_rng_state):
+                return func(*primals)
+
+    PhiloxStateTracker.reset()
+
+    if trace_joint:
+        # Get the current seed and offset to setup tracing.
+        fwd_seed, fwd_base_offset = RNGStateHelper.get_torch_state_as_tuple()
```

Comment:
Can't you do this before fakeification happens so that you don't have to do it by hand here?